xapian-letor/ranker/ranker.cc

```diff
@@ -354,6 +354,9 @@ Ranker::labelcomparer(const FeatureVector & firstfv, const FeatureVector& second
 std::vector<Xapian::docid>
 Ranker::rank(const Xapian::MSet & mset, const string & model_key, const Xapian::FeatureList & flist)
 {
+    if (mset.empty()) {
+	throw Xapian::InvalidArgumentError("Empty MSet. No documents could be retrieved with the given Query.\n");
```

Comment:
Here we can just `return std::vector<Xapian::docid>()` - that's a valid reranking of an empty `MSet` (and indeed the only possible reranking).

As a general point though, exception messages should not include a newline at the end.